internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -216,53 +210,48 @@ func resourceCassandraDatacenterRead(d *pluginsdk.ResourceData, meta interface{}
 }
 
 func resourceCassandraDatacenterUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cosmos.CassandraDatacentersClient
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.CassandraDatacenterID(d.Id())
+	id, err := managedcassandras.ParseDataCenterID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	body := documentdb.DataCenterResource{
-		Properties: &documentdb.DataCenterResourceProperties{
-			DelegatedSubnetID:  utils.String(d.Get("delegated_management_subnet_id").(string)),
-			NodeCount:          utils.Int32(int32(d.Get("node_count").(int))),
+	body := managedcassandras.DataCenterResource{
+		Properties: &managedcassandras.DataCenterResourceProperties{
+			DelegatedSubnetId:  utils.String(d.Get("delegated_management_subnet_id").(string)),
+			NodeCount:          utils.Int64(int64(d.Get("node_count").(int))),
 			DataCenterLocation: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
 			DiskSku:            utils.String(d.Get("disk_sku").(string)),
 		},
 	}
 
 	if v, ok := d.GetOk("backup_storage_customer_key_uri"); ok {
-		body.Properties.BackupStorageCustomerKeyURI = utils.String(v.(string))
+		body.Properties.BackupStorageCustomerKeyUri = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("base64_encoded_yaml_fragment"); ok {
 		body.Properties.Base64EncodedCassandraYamlFragment = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("managed_disk_customer_key_uri"); ok {
-		body.Properties.ManagedDiskCustomerKeyURI = utils.String(v.(string))
+		body.Properties.ManagedDiskCustomerKeyUri = utils.String(v.(string))
 	}
 
-	future, err := client.CreateUpdate(ctx, id.ResourceGroup, id.CassandraClusterName, id.DataCenterName, body)
-	if err != nil {
+	if err := client.CassandraDataCentersCreateUpdateThenPoll(ctx, *id, body); err != nil {
 		return fmt.Errorf("updating %q: %+v", id, err)
 	}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting on update for %q: %+v", id, err)
-	}
-
 	// Issue: https://github.com/Azure/azure-rest-api-specs/issues/19078
 	// There is a long running issue on updating this resource.
 	// The API cannot update the property after WaitForCompletionRef is returned.
 	// It has to wait a while after that. Then the property can be updated successfully.
 	stateConf := &pluginsdk.StateChangeConf{
```

Comment:
is this still needed with `hashicorp/go-azure-sdk`?

Reply:
After tested, I found we still need it otherwise it would fail.